src/plugins/translate/plugin.translate.js

```diff
@@ -60,8 +60,25 @@ export class TranslatePlugin extends BookReaderPlugin {
       }
 
       const pageElement = pageContainer.$container[0];
-      this.translateRenderedLayer(pageElement);
+      await this.translateRenderedLayer(pageElement);
     });
+
+    /**
+     * @param {*} ev
+     * @param {object} eventProps
+    */
+    this.br.on('pagevisible', (_, {pageContainerEl}) => {
+      const translationLayerEl = Array.from(pageContainerEl.childNodes).filter((ele) => {
+        return ele.classList.contains('BRtranslateLayer');
+      });
+      if (translationLayerEl) {
+        const translatedParagraphs = this.getTranslatedParagraphsOnPage(pageContainerEl);
+        for (const ele of translatedParagraphs) {
+          this.fitVisiblePage(ele);
+        }
+      }
```

Comment:
This can be done more simply with querySelectorAll, and removes the necessity for the helper function!

```js
for (const paragEl of pageContainerEl.querySelectorAll('.BRtranslateLayer > .BRparagraphElement')) {
    this.fitVisiblePage(paragEl);
}
```